src/Workspaces/Core/Portable/Workspace/Host/PersistentStorage/IPersistentStorageService.cs

```diff
@@ -18,7 +19,8 @@ public interface IPersistentStorageService : IWorkspaceService
 
     internal interface IPersistentStorageService2 : IPersistentStorageService
     {
-        IPersistentStorage GetStorage(Solution solution, bool checkBranchId);
-        IPersistentStorage GetStorage(Workspace workspace, SolutionKey solutionKey, bool checkBranchId);
+        ValueTask<IPersistentStorage> GetStorageAsync(Solution solution, CancellationToken cancellationToken);
```

Comment:
❔ Why not just add this method to `IPersistentStorageService`?

Reply:
Could do that. But it would affect the public surface area of roslyn here. If you're ok with that, I'll move it.